### PR TITLE
[intel-npu] Updating npu-device properties in documentation

### PR DESCRIPTION
**Pull request by csoka**
### Details:
 - adding newly supported properties to npu-device documentation
 - addig section describing compilation-mode-params to npu-device documentation

### Tickets:
 - *ticket-id*

### CODE REVIEW

**Comment by csoka** [comment]
please review, as i am not confident with the rst code. thanks!

File: docs/articles_en/openvino-workflow/running-inference/inference-devices-and-modes/npu-device.rst [hunk 1]
```diff
@@ -155,6 +162,53 @@ offer a limited set of supported OpenVINO features.
    based on the performance mode is **4 for THROUGHPUT** and **1 for LATENCY**.
    The default mode for the NPU device is LATENCY.
 
+**ov::intel_npu::compilation_mode_params**
+   ov::intel_npu::compilation_mode_params is an NPU-specific property that allows to 
+   control model compilation for NPU.  
+   .. note::
+      The functionality is in experimental stage currently, can be a subject for 
+      deprecation and may be replaced with generic OV API in future OV releases.
```
Comment (msmykx-intel) [inline docs/articles_en/openvino-workflow/running-inference/inference-devices-and-modes/npu-device.rst:170]:
```suggestion
**ov::intel_npu::compilation_mode_params**

``ov::intel_npu::compilation_mode_params`` is an NPU-specific property that allows to
control model compilation for NPU.

.. note::

   The functionality is in experimental stage currently, can be a subject for
   deprecation and may be replaced with generic OV API in future OV releases.
```

File: docs/articles_en/openvino-workflow/running-inference/inference-devices-and-modes/npu-device.rst [hunk 2]
```diff
@@ -155,6 +162,53 @@ offer a limited set of supported OpenVINO features.
...
+      :header-rows: 1
+
+      * - **Value**
+        - **Description**
+      * - 0
+        - Reduced subset of optimization passes. Smaller compile time.
+      * - 1
+        - **Default.** Balanced performance/compile time.
+      * - 2
+        - Prioritize performance over compile time that may be an issue.
```
Comment (msmykx-intel) [inline docs/articles_en/openvino-workflow/running-inference/inference-devices-and-modes/npu-device.rst:187]:
```suggestion
Following configuration options are supported:

**optimization-level**

Defines a preset of optimization passes to be applied during compilation.

.. list-table::
   :widths: 10 200
   :header-rows: 1

   * - **Value**
     - **Description**
   * - 0
     - Reduced subset of optimization passes. Smaller compile time.
   * - 1
     - **Default.** Balanced performance/compile time.
   * - 2
     - Prioritize performance over compile time that may be an issue.
```

File: docs/articles_en/openvino-workflow/running-inference/inference-devices-and-modes/npu-device.rst [hunk 3]
```diff
@@ -155,6 +162,53 @@ offer a limited set of supported OpenVINO features.
...
+        - Prioritize performance over power efficiency. 
+   
+   .. tab-set::
+
+      .. tab-item:: Usage example
+
+         .. code-block::
+            map<str, str> config = {ov::intel_npu::compilation_mode_params.name(), ov::Any("optimization-level=1 performance-hint-override=latency")}; 
+
+            compile_model(model, config);
```
Comment (msmykx-intel) [inline docs/articles_en/openvino-workflow/running-inference/inference-devices-and-modes/npu-device.rst:211]:
```suggestion
**performance-hint-override**

An extension for LATENCY mode being specified using ``ov::hint::performance_mode``
Has no effect for other ``ov::hint::PerformanceMode`` hints.

.. list-table::
   :widths: 10 200
   :header-rows: 1

   * - **Value**
     - **Description**
   * - efficiency
     - **Default.** Balanced performance and power consumption.
   * - latency
     - Prioritize performance over power efficiency.

.. tab-set::

   .. tab-item:: Usage example

      .. code-block::

         map<str, str> config = {ov::intel_npu::compilation_mode_params.name(), ov::Any("optimization-level=1 performance-hint-override=latency")};

         compile_model(model, config);
```

**Review by msmykx-intel — APPROVED** [review]

**Pull request merged**